src/containers/Signin/signinActions.js

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+
+//actions creators
+export function fetchSigninToken(username, password) {
+    return  {
+        type: "FETCH_USERINFO",
+        payload: axios({ 
+            method: "post",
+            url: "https://frontend-challenge-2.herokuapp.com/rest-auth/login/",
+            data: {
+                username: username,
+                password: password
+            }
+        })
+        .then(function (response) {
+            var signinToken = response.data.key;
+            localStorage.setItem('tokenkey',signinToken);
+            var signinTokenStorage = localStorage.getItem('tokenkey');
+            window.location = '/home';
+        })
+        .catch(function (error) {
+            alert('Unable to log in with provided credentials, check email or password');
```

Comment:
As I told you in person and as mentioned earlier, you need to show mesage based on particular cases.